app/gui/src/controller/searcher/component.rs

```diff
@@ -54,6 +55,31 @@ pub enum Order {
 
 
 
+// ============
+// === Data ===
+// ============
+
+/// Contains detailed data of a [`Component`]. The storage of the details differs depending on
+/// where the data originates from (either from the [`suggestion_database`] or from a
+/// [`hardcoded::Snippet`]).
+#[derive(Clone, CloneRef, Debug)]
+pub enum Data {
+    /// A component from the [`suggestion_database`].
+    FromDatabase {
+        /// The ID of the component in the [`suggestion_database`].
+        id:    Immutable<Id>,
+        /// The component's entry in the [`suggestion_database`].
+        entry: Rc<suggestion_database::Entry>,
+    },
+    /// A virtual component containing a hardcoded snippet of code.
```

Comment:
Please add the more explicit info how the "component from snippet" work: that the snippet has defined return type, and accepting the component shall just put the snippet at the end of searcher input.

Reply:
Based on our discussion, I did not mention return type, as it is not currently used. I also did not mention "end of searcher input", but instead used similar phrasing as was used in the documentation of `Component` before.